gulpfile.js/tasks/server.js

```diff
@@ -1,11 +1,84 @@
 'use strict'
 
 const gulp = require('gulp')
+const portfinder = require('portfinder')
+const gutil = require('gulp-util')
 const browserSync = require('browser-sync')
-const config = require('../config').browserSync
+const colors = require('colors')
+const config = require('../config')
+const assetsPathV3 = `http://localhost:9032/public/v3-SNAPSHOT/`
+const assetsPathV4 = `http://localhost:9032/public/v4-SNAPSHOT/`
 
 gulp.task('server', () => {
-  config.forEach((conf, i) => {
-    browserSync.create(`${i}`).init(conf)
-  })
+  if (gutil.env.version === 'v3') {
+    browserSync.create().init(config.browserSync.assets, afStarted)
+    startServer('patternLibrary')
+      .then(() => {
+        startServer('componentLibrary')
+      })
+  } else {
+    browserSync.create().init(config.browserSync.assets, afStarted)
+    startServer('patternLibrary')
+  }
 })
+
+const startServer = (serverName) => {
+  const serverPath = config[serverName].dest
+  portfinder.basePort = config[serverName].basePort
+  let serverConfig
+  return new Promise((resolve, reject) => {
+    const hasStarted = () => {
+      message(`${config[serverName].friendlyName} started on http://localhost:${serverConfig.port}/`)
+      resolve()
+    }
+    try {
+      getServerConfig(serverPath).then((config) => {
+        serverConfig = config
+        browserSync.create(serverPath).init(config, hasStarted)
+      })
+    } catch (err) {
+      reject(err)
+    }
+  })
+}
+
+const getServerConfig = (serverPath) => {
+  return new Promise((resolve, reject) => {
+    portfinder.getPort((err, port) => {
+      if (err) {
+        reject(err)
+      } else {
+        resolve({
```

Comment:
This stuff should live in config really.

There's a lot going on in this file that a simpler config object could help solve.